src/composables/useTable.ts

```diff
@@ -0,0 +1,170 @@
+      align: 'left',
+      field: 'name',
+      sortable: true,
+    },
+    {
+      name: 'team',
+      required: true,
+      label: i18n.global.t('results.labelTeam'),
+      align: 'left',
+      field: 'team',
+      sortable: true,
+    },
+    {
+      name: 'organization',
+      required: true,
+      label: i18n.global.t('results.labelOrganization'),
+      align: 'left',
+      field: 'organization',
+      sortable: true,
+    },
+    {
+      name: 'category',
+      required: true,
+      label: i18n.global.t('results.labelCategory'),
+      align: 'left',
+      field: 'category',
+      sortable: true,
+    },
+    {
+      name: 'city',
+      required: true,
+      label: i18n.global.t('results.labelCity'),
+      align: 'left',
+      field: 'city',
+      sortable: true,
+    },
+  ];
+
+  const searchQuery = ref('');
+  const filterQuery = ref('');
+
+  /**
+   * Creates an object containing the search query and the filter query.
+   */
+  const filterCompound = computed((): FilterMethodInput => {
+    return {
+      search: searchQuery.value,
+      filter: filterQuery.value,
+    };
+  });
+
+  /**
+   * Provides filter functionality
+   * Upon typing, find strings which contain query entered into the filter
+   * Function must match QTable filterMethod interface
+   * @param rows TableRow[]
+   * @param terms FilterMethodInput
+   * @param cols TableColumn[]
+   * @param cellValue (col: TableColumn, row: TableRow) => string
+   **/
+  const filterMethod = (
+    rows: readonly TableRow[],
+    terms: FilterMethodInput,
+    cols: readonly TableColumn[],
+    cellValue: (col: TableColumn, row: TableRow) => string,
+  ): readonly TableRow[] | undefined => {
+    const { search, filter } = terms;
+    if (!search && !filter) {
+      return rows;
+    }
+    if (!filter) {
+      defaultFilter(search);
+    }
+    if (!search) {
+      defaultFilter(filter);
+    }
+    // both filter options are selected
+    const lowerTerms = [search, filter].map((query) => query.toLowerCase());
+    return rows.filter((row) => {
+      // combine conditions into an && operator
+      return (
+        !!isMatch({ cols, row, term: lowerTerms[0] }) &&
+        !!isMatch({ cols, row, term: lowerTerms[1] })
+      );
+    });
+
+    /**
+     * Default filter function based on QTable source code
+     * @param query string
+     */
+    function defaultFilter(query: string): TableRow[] {
+      const lowerTerms = query ? query.toLowerCase() : '';
+      return rows.filter((row) => !!isMatch({ cols, row, term: lowerTerms }));
+    }
+
+    /**
+     * Checks if a row matches the search query
+     * @param row TableRow
+     * @param term string
+     */
+    function isMatch({
+      cols,
+      row,
+      term,
+    }: {
+      cols: readonly TableColumn[];
+      row: TableRow;
+      term: string;
+    }): boolean {
+      return cols.some((col) => {
+        const val = cellValue(col, row) + '';
```

Comment:
Why we need concatenate row value string with `''` string?

Reply:
For better readibility you can use for  type conversion `.toString()` method.

Reply:
Fixed with commit https://github.com/auto-mat/ride-to-work-by-bike-frontend/pull/394/commits/d647fe43c4d0a77219be04be6d7bb8a855ef21e4.